### DISCUSSION TITLE
Customizable standard button (not apps)

### DISCUSSION THREAD

**Discussion by qwerty13547** [PRProd/HA-Firemote#462] [Ideas]
Hi Doug!

You save me yesterday when my Shield remote die on me. Really great work on the remote functionality. 👍

I read about the possibility to make "long-press" button by sacify one of the app-launch buttons. That is a good solution for that particular issue. However, Shield has more functions that will sacrify even more app-lauch buttons and I would like to keep them if possible.

So my question are if is it possible to add more customizable standard buttons and place them amongst the other play like buttons.
Functions that would be interesting:
- Either buttons or a "outer-ring" for long press in each directions. Example fast-forwarding for a particular spot in a youtube clip. 
- Button for long click select. Example: Getting a menu for a clip in youtube.  
- Button for dubble click. Example: opens a menu to close an app in Shield. 

And the unobtainable wish of total customization. Drag-n-drop each button where-ever, change size and icon to fit each device that I have dashboard for. It might already be possible to code but if it is in your easy to understand GUI I would be so much better.  Lots of coffees if you manage that 😉

If my english not is understandale let me know and I will show a image of how i mean.

Cheers!
Johan

**Comment by PRProd** [comment]
Greetings Johan @qwerty13547!

Who needs batteries when you have Firemote?  😄 I'm happy to hear that my humble little card saved you!

I do have some potentially good news regarding "long press".  I haven't had a new [release](https://github.com/PRProd/HA-Firemote/releases) for a while now because I've been working on a beta version of _long press_ built right into the Firemote.  (mentioned in #378) It has been trickier than I ever expected, but I am making progress!  Keep your eyes open for that in the near future.

As far as _double click_ goes, that should already work!  If it's not working properly for you, please let me know some additional details about the issue so I can take a look.

In the meantime, have you tried using **App Launcher 1** or **App Launcher 2** remote styles?
![image](https://github.com/user-attachments/assets/729cc90f-fb07-4a08-87fe-6fc41cd1441a)
![image](https://github.com/user-attachments/assets/d9bd8bba-6e34-429e-a600-70d4b935ded1)
These can be great alternatives OR be used alongside your regular looking Shield remote to provide the extra buttons that you might want.  If you haven't looked yet, check out the [Let's See Your Dashboards! #129](https://github.com/PRProd/HA-Firemote/discussions/129) discussion to see some examples of how to do that.

If those alternatives and/or the idea of upcoming _long press_ functionality doesn't quite satisfy your request, I'd love to see an example of what you might like!  Feel free to sketch out a few ideas and attach them here so we can talk about them!

I REALLY appreciate your kind comments about how easy the GUI is to use and understand!  Thank you for telling me!  I love to hear those kinds of things.  I'm not sure if I would ever be able to make this as awesome as your dream idea of drag and drop, but that would be pretty cool if I did!  Thanks for the idea!

You mentioned that English isn't your primary language.  Do you speak a language that isn't available for Firemote yet?  [Check out the list of supported languages here.](https://github.com/PRProd/HA-Firemote/wiki/#which-languages-are-currently-supported)  If so, would you be willing to contribute to the Firemote project by opening a [New Language Support Request](https://github.com/PRProd/HA-Firemote/issues/new?assignees=PRProd&labels=Region+Issue%2Cenhancement&projects=&template=New_Language_Support.yaml&title=%5BNew+Language+Support%5D%3A+)?

Thanks for reaching out!  I hope to hear back from you soon!

-Doug

**Reply by qwerty13547** [reply]
Hi Doug,
Sorry for my lack of reply. Have been away on vacation.

I will try to make use of App Launcher 2. It might solve my issue. As I use both mobile and Pad I might need to make 2 different to suit both devices. When I am done I will post them in the discussion.

I have submitted a new language request. 👍

**Comment by PRProd** [comment]
@qwerty13547 ~

The long awaited beta release for long press support is now here!  Take a look at discussion #477 for details!

-Doug

**Reply by qwerty13547** [reply]
Hi, 

I have replied in the beta group regarding longpress. 
Double click is not relevant anymore as the function I needed was for Nvidia and was acctually already there (Switch app). It worked just fine   👍

Have a nice evening 🙏
